theano/scalar/basic.py

```diff
@@ -3659,7 +3659,7 @@ def __init__(self, inputs, outputs):
         self.nin = len(inputs)
         self.nout = len(outputs)
         self.init_fgraph()       # self.fgraph
-
+        self.init_c_code()
```

Comment:
The reason for that : https://github.com/Theano/Theano/pull/4749#issuecomment-252755870 
This Op compares with `_c_code` and the Op doesn't have  the attribute until we call this. When we don't have this, I get this error : 
```
ERROR (theano.gof.opt): SeqOptimizer apply <theano.tensor.opt.FusionOptimizer object at 0x105774610>
ERROR (theano.gof.opt): Traceback:
ERROR (theano.gof.opt): Traceback (most recent call last):
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 235, in apply
    sub_prof = optimizer.optimize(fgraph)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 87, in optimize
    ret = self.apply(fgraph, *args, **kwargs)
  File "/Users/Ramana/projects/theano/Theano/theano/tensor/opt.py", line 7156, in apply
    reason=self.__class__.__name__)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/toolbox.py", line 340, in replace_all_validate
    fgraph.replace(r, new_r, reason=reason, verbose=False)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 531, in replace
    self.change_input(node, i, new_r, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 471, in change_input
    self.__import_r__(new_r, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 372, in __import_r__
    self.__import__(variable.owner, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 435, in __import__
    self.execute_callbacks('on_import', node, reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 611, in execute_callbacks
    fn(self, *args, **kwargs)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 561, in on_import
    self.process_node(fgraph, node)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 675, in process_node
    if inputs_match and node.op == candidate.op:
  File "/Users/Ramana/projects/theano/Theano/theano/gof/utils.py", line 196, in __eq__
    tuple(getattr(other, a) for a in props))
  File "/Users/Ramana/projects/theano/Theano/theano/scalar/basic.py", line 3775, in __eq__
    return (self._c_code == other._c_code)
AttributeError: 'Composite' object has no attribute '_c_code'
```